felix/daemon/daemon.go

```diff
@@ -369,14 +369,31 @@ configRetry:
 	}
 
 	// We may need to temporarily disable encrypted traffic to this node in order to connect to Typha
+	var wireguardPeersToValidate set.Set
 	if configParams.WireguardEnabled {
-		err := bootstrapWireguard(configParams, v3Client)
+		var err error
+		wireguardPeersToValidate, err = bootstrapWireguard(configParams, v3Client)
 		if err != nil {
 			time.Sleep(2 * time.Second) // avoid a tight restart loop
 			log.WithError(err).Fatal("Couldn't bootstrap WireGuard host connectivity")
 		}
 	}
 
+	// If we have wireguard peers that we need to validate against our typha nodes then do that now. We do this because
```

Comment:
Does it matter that this logic isn't within a "If typha enabled" branch? i.e., we'll run this even if connecting directly to the datastore?

Reply:
I think that it probably does - we don't distinguish between "no typha peers because typha is disabed" and "no typha peers because typha is enabled but there are no valid peers", right?

Reply:
Argh quite correct.  When I was implementing this I had in my head about that... basically don't do this if there are no typhas to begin with.

That logic vanished somewhere between my brain and the keyboard.